app/cli/internal/action/attestation_push.go

```diff
@@ -107,6 +109,9 @@ func (action *AttestationPush) Run(runtimeAnnotations map[string]string) (*Attes
 
 	action.Logger.Debug().Msg("validation completed")
 
+	// Indicate that we are done with the attestation
+	action.c.CraftingState.Attestation.FinishedAt = timestamppb.New(time.Now())
```

Comment:
The finishedAt timestamp is now in the attestation state. This simplifies testing by making rendering idempotent.